File: src/types/salarySlip.ts
export type ExtractionMethod = 'NOT_IDENTIFIED' | 'TEXT_EMBEDDED' | 'OCR' | 'MANUAL';

export type MatchStatus =
  | 'UNMATCHED'
  | 'NOT_IDENTIFIED'
  | 'IDENTIFIED'
  | 'PARTIALLY_IDENTIFIED'
  | 'TEXT_EXTRACTION_FAILED'
  | 'DUPLICATE_CONTENT'
  | 'EXACT_MATCH'
  | 'STRONG_MATCH'
  | 'POSSIBLE_MATCH'
  | 'NO_MATCH'
  | 'CONFLICT'
  | 'MANUAL_REVIEW'
  | 'MANUALLY_CONFIRMED'
  | 'MANUALLY_REJECTED'
  | 'READY'
  | 'REVIEW_REQUIRED'
  | 'CONFIRMED'
  | 'REJECTED'
  | 'FILE_MISSING';

export type ApprovalStatus = 'PENDING' | 'APPROVED' | 'REJECTED';
export type OcrStatus =
  | 'NOT_REQUIRED'
  | 'PENDING'
  | 'RUNNING'
  | 'COMPLETED'
  | 'COMPLETED_WITH_WARNINGS'
  | 'UNAVAILABLE'
  | 'RENDER_FAILED'
  | 'ENGINE_ERROR'
  | 'EMPTY_RESULT'
  | 'TIMEOUT'
  | 'FAILED';

export type DocumentType = 'SALARY_SLIP' | 'POSSIBLE_SALARY_SLIP' | 'NOT_SALARY_SLIP' | 'UNKNOWN';

export interface SalarySlip {
  id: string;
  filePath: string;
  fileName: string;
  fileHash: string;
  detectedEmployeeId?: string;
  detectedName?: string;
  detectedPhone?: string;
  detectedEmail?: string;
  extractionMethod: ExtractionMethod;
  extractedText?: string;
  matchConfidence: number;
  matchStatus: MatchStatus;
  duplicateOfId?: string;
  ocrConfidence?: number;
  ocrProcessedAt?: string;
  ocrError?: string;
  matchedEmployeeId?: string;
  matchReason?: string;
  matchedAt?: string;
  reviewedAt?: string;
  reviewedBy?: string;
  reviewNote?: string;
  month?: string;
  year?: string;
  approvalStatus: ApprovalStatus;
  ocrStatus: OcrStatus;
  documentType?: DocumentType;
  documentConfidence?: number;
  ocrAttemptCount?: number;
  ocrPageCount?: number;
  ocrProcessingTimeMs?: number;
  createdAt: string;
  updatedAt: string;
}

export interface ScanError {
  path: string;
  errorKind: string;
  message: string;
}

export interface DiscoveredFile {
  filePath: string;
  fileName: string;
  fileExtension: string;
  fileSize: number;
  modifiedAt: string;
  fileHash: string;
  month?: string;
  year?: string;
}

export interface FolderScanDiagnostics {
  selectedPath: string;
  displayName: string;
  exists: boolean;
  isDirectory: boolean;
  readable: boolean;
  pdfCount: number;
  directoriesScanned: number;
  filesScanned: number;
  databaseRecords: number;
  scanErrors: ScanError[];
  files: DiscoveredFile[];
}

export interface ScanSummary {
  totalScanned: number;
  pdfCount: number;
  newCount: number;
  updatedCount: number;
  unchangedCount: number;
  duplicateCount: number;
  folderPath: string;
  displayName?: string;
  directoriesScanned?: number;
  filesScanned?: number;
  scanErrors?: (string | ScanError)[];
  files?: DiscoveredFile[];
  slips: SalarySlip[];
}

export interface ExtractionSummary {
  total: number;
  processed: number;
  identified: number;
  partiallyIdentified: number;
  notIdentified: number;
  failed: number;
  skipped: number;
  slips: SalarySlip[];
}

export interface OcrBatchSummary {
  total: number;
  processed: number;
  identified: number;
  partiallyIdentified: number;
  notIdentified: number;
  failed: number;
  skipped: number;
  slips: SalarySlip[];
}

File: src-tauri/src/models/salary_slip.rs
use serde::{Deserialize, Serialize};
use crate::filesystem::{DiscoveredFile, ScanError};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SalarySlip {
    pub id: String,
    pub file_path: String,
    pub file_name: String,
    pub file_hash: String,
    pub detected_employee_id: Option<String>,
    pub detected_name: Option<String>,
    pub detected_phone: Option<String>,
    pub detected_email: Option<String>,
    pub extraction_method: String,
    pub extracted_text: Option<String>,
    pub match_confidence: f64,
    pub match_status: String,
    pub duplicate_of_id: Option<String>,
    pub ocr_confidence: Option<f64>,
    pub ocr_processed_at: Option<String>,
    pub ocr_error: Option<String>,
    pub matched_employee_id: Option<String>,
    pub match_reason: Option<String>,
    pub matched_at: Option<String>,
    pub reviewed_at: Option<String>,
    pub reviewed_by: Option<String>,
    pub review_note: Option<String>,
    pub month: Option<String>,
    pub year: Option<String>,
    pub approval_status: String,
    pub ocr_status: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ScanSummary {
    pub total_scanned: usize,
    pub pdf_count: usize,
    pub new_count: usize,
    pub updated_count: usize,
    pub unchanged_count: usize,
    pub duplicate_count: usize,
    pub folder_path: String,
    pub display_name: String,
    pub directories_scanned: usize,
    pub files_scanned: usize,
    pub scan_errors: Vec<ScanError>,
    pub files: Vec<DiscoveredFile>,
    pub slips: Vec<SalarySlip>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExtractionSummary {
    pub total: usize,
    pub processed: usize,
    pub identified: usize,
    pub partially_identified: usize,
    pub not_identified: usize,
    pub failed: usize,
    pub skipped: usize,
    pub slips: Vec<SalarySlip>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct OcrBatchSummary {
    pub total: usize,
    pub processed: usize,
    pub identified: usize,
    pub partially_identified: usize,
    pub not_identified: usize,
    pub failed: usize,
    pub skipped: usize,
    pub slips: Vec<SalarySlip>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct BulkConfirmResult {
    pub confirmed_count: usize,
    pub skipped_count: usize,
    pub skipped_reasons: Vec<String>,
    pub slips: Vec<SalarySlip>,
}
